doc/internal/consensus/Consensus-Protocol.md

```diff
@@ -20,6 +20,80 @@ Raft is an algorithm to handle replicated log, and we maintain the "log" of the
 * Request retries are done in case of network failures.
 * Raft does not assume network preserves ordering of the packets.
 * The `raft/raft.go` file has all the state parameters and general functions that raft uses.
+* A raft server is implemented as:
+```
+type simpleServer struct {
+	node          *Node
+	cluster       Cluster
+	onReplication ReplicationHandler
+	log           zerolog.Logger
+}
+
+type Node struct {
+	State string
+
+	PersistentState     *PersistentState
+	VolatileState       *VolatileState
+	VolatileStateLeader *VolatileStateLeader
+}
+
```

Comment:
why is there implementation in a design doc?

Reply:
Well if I add the API, they struct's would be a good reference to have, was my thought.

Reply:
Keep in mind that you are writing a design doc, not an implementation guide. Such a concrete struct may have heavy implications on the rest of the implementation. The point of a design doc is, to just outline the components and how they interact, and leave the concrete implementation to the developer.